state-chain/pallets/cf-funding/src/lib.rs

```diff
@@ -529,11 +540,20 @@ pub mod pallet {
 
 			// If the address is still restricted, we update the restricted balances again.
 			if RestrictedAddresses::<T>::contains_key(address) {
+				let restricted_balance_for_address =
+					PendingRedemptionsRestrictedBalance::<T>::take(&account_id).unwrap_or_default();
+				// Ensure that the balance we set for the restricted address is not more than the
+				// balance we have recorded during redemption for the address.
+				let restricted_amount = if amount >= restricted_balance_for_address {
```

Comment:
We don't expect this to ever be the case (and IMO adding this conditional clause suggests that it might legitimately occur). I would prefer a `debug_assert` here to ensure that the condition holds in our unit tests.